src/integrity-checker.js

```diff
@@ -214,7 +213,16 @@ export default class InstallationIntegrityChecker {
     let integrityMatches;
     if (expected) {
       integrityMatches = this._compareIntegrityFiles(actual, expected);
-      if (flags.checkFiles && expected.files.length > 0) {
+      if (flags.checkFiles && expected.files.length === 0) {
+        // edge case handling - --check-fies is passed but .yarn-integrity does not contain any files
+        // check and fail if there are file in node_modules after all.
+        const actualFiles = [];
+        await this._getFilePaths(loc.locationFolder, actualFiles);
```

Comment:
Do you think it would be better this way?
_getFilePaths could have a return value, I actually considered making it a non recursive function but did not come up with a more readable code.

Do you want to suggest a non recursive _getFilePaths implementation?